src/test/java/uk/gov/pay/api/clients/ExternalServiceClientTest.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.pay.api.clients;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.slf4j.MDC;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ExternalServiceClientTest {
+
+    private ExternalServiceClient externalServiceClient;
+
+    @Mock
+    private Client client;
+    @Mock
+    private WebTarget webTarget;
+    @Mock
+    private Invocation.Builder builder;
+    @Captor
+    private ArgumentCaptor<MultivaluedMap<String, Object>> builderArgumentCaptor;
+
+    @Before
+    public void setUp() {
+        externalServiceClient = new ExternalServiceClient(client);
+        when(client.target(anyString())).thenReturn(webTarget);
+        when(webTarget.request()).thenReturn(builder);
+        when(builder.headers(any(MultivaluedMap.class))).thenReturn(builder);
+        when(builder.accept(MediaType.APPLICATION_JSON)).thenReturn(builder);
+    }
+    
+    @Test
+    public void assert_x_request_id_in_request_header_for_get() {
```

Comment:
The naming of this test doesn't fit with our conventions

Reply:
I'm not we have coding standards for these anywhere ;) Alex doesn't appear to have minded when I did the same thing at https://github.com/alphagov/pay-connector/pull/1907/files.
I do think underscores are much more readable than camel case. 
Philosophically I think I'm paid to write readable code that works, rather than blindly follow conventions :)

Reply:
You make a fair point.
My counter argument would be that it's tidier to follow the unwritten conventions as it doesn't create the mental overhead of deciding which style to use. And if you come to expect things are a certain way it is a bit jarring to see it different in some places.
It's not a hill I'm going to die on though, so won't withhold approval over this.

Reply:
There's an existing proliferation of underscores in https://github.com/alphagov/pay-publicapi/blob/master/src/test/java/uk/gov/pay/api/validation/PaymentSearchValidatorTest.java too. I reckon people realise camel case becomes unreadable after a while.

Reply:
Someone mentioned my name!

In general, all things being equal, I’ve become convinced over the years that `snake_case_is_easier_to_read` than `camelCaseIsToRead`.

However, in Java, `camelCaseIsKing` (`EXCEPT_FOR_CONSTANT_NAMES`) and there is a very, very strong convention that method names are camel case. This convention is so strong that I think it should probably considered an implied default in any Java coding standard unless there is guidance to the contrary.

As it happens, we do use underscores in some test method names and have done for longer than I’ve been here. An example would be `makeAppleStrudel_withIcingSugarOnTop_whenGuestsArrive()`. Here, the method name is actually a Frankenstein combination of camel case and snake case, with underscores used to split up the different aspects of the test: the expected result, a property of the result which perhaps differentiates this test from similar tests, and the action that causes the result. I’m personally opposed to using underscores in this way because it conflicts with general Java conventions. But other people aren’t, so our [PMD configuration allows underscores in JUnit test method names](https://github.com/alphagov/pay-publicapi/blob/master/ruleset.xml#L94-L95).

I haven’t seen full snake case used for tests before (I missed it in the other review, sorry!). While it is readable, I’m not generally in favour of this because it’s very un-Java-like. I don’t think there would be a huge readability difference if the test method name were `assertXRequestIdInRequestHeaderForGet()`.

This feels like a good thing to bring up at the next tech team meeting to see if we can find a consensus among the team.

Reply:
>This feels like a good thing to bring up at the next tech team meeting to see if we can find a consensus among the team.

Why not we do that and not delay this PR being merged. Of more importance is that we can start tracing requests properly in production.

Reply:
I never meant to suggest that we should delay the merging of this PR.